pkg/k8s/watchers/pod.go

```diff
@@ -123,10 +123,16 @@ func (k *K8sWatcher) createPodController(getter cache.Getter, fieldSelector fiel
 func (k *K8sWatcher) podsInit(k8sClient kubernetes.Interface, asyncControllers *sync.WaitGroup) {
 	var once sync.Once
 	watchNodePods := func() chan struct{} {
-		// Only watch for pod events for our node.
+		// Only watch for pod events for our node. Watch all pods when EnableEgressGateway.
+		// Ideally only gateway nodes need to watch all pods, once there is a way to identify
+		// gateway nodes from normal nodes.
+		nodeSelector := fields.ParseSelectorOrDie("spec.nodeName=" + nodeTypes.GetName())
+		if option.Config.EnableEgressGateway {
+			nodeSelector = fields.Everything()
+		}
```

Comment:
This will not scale. I strongly advise to watch for CEPs instead because it contains all of the information required for egress policy manager and it has better scalability than pods.

Reply:
This is a good point. We do have that item planned after v1.10. But since it's brought up here. I feel it's better to address it now while I can.
This is updated in the newest version. The pod/ns watchers are replaced by the cilium endpoint watcher logic now.